File: src/main/java/org/micoli/php/PhpCompanionProjectStartup.java
package org.micoli.php;

import com.intellij.openapi.diagnostic.Logger;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.startup.ProjectActivity;
import kotlin.Unit;
import kotlin.coroutines.Continuation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class PhpCompanionProjectStartup implements ProjectActivity {
    private static final Logger LOG = Logger.getInstance(PhpCompanionProjectStartup.class);

    @Override
    public @Nullable Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
        LOG.info("Starting PhpCompanionProjectService for project: " + project.getName());
        PhpCompanionProjectService.getInstance(project);
        return Unit.INSTANCE;
    }
}

File: src/main/kotlin/org/micoli/php/PhpCompanionProjectStartup.kt
package org.micoli.php

import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.project.Project
import com.intellij.openapi.startup.ProjectActivity

class PhpCompanionProjectStartup : ProjectActivity {
    override suspend fun execute(project: Project) {
        LOG.info("Starting PhpCompanionProjectService for project: ${project.name}")
        PhpCompanionProjectService.getInstance(project)
    }

    companion object {
        private val LOG = Logger.getInstance(PhpCompanionProjectStartup::class.java.getSimpleName())
    }
}
